data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/PluginConfigsJsonSchemaConverter.java

```diff
@@ -0,0 +1,119 @@
+package org.opensearch.dataprepper.schemas;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.victools.jsonschema.generator.OptionPreset;
+import com.github.victools.jsonschema.generator.SchemaVersion;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.configuration.ConditionalRoute;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.sink.Sink;
+import org.opensearch.dataprepper.model.source.Source;
+import org.reflections.Reflections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PluginConfigsJsonSchemaConverter {
+    private static final Logger LOG = LoggerFactory.getLogger(PluginConfigsJsonSchemaConverter.class);
+    static final String DOCUMENTATION_LINK_KEY = "documentation";
+    static final String PLUGIN_NAME_KEY = "name";
+    static final String PLUGIN_DOCUMENTATION_URL_FORMAT =
+            "https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/%s/%s/";
```

Comment:
Should these links be the full path?

We typically use a pattern like:

```
{{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/%s/%s/
```

e.g.

https://github.com/opensearch-project/documentation-website/blob/a673b24b7b02a2969795b1e9ee379e3533611c00/_data-prepper/common-use-cases/s3-logs.md?plain=1#L133